### PR TITLE
Update promise documentation

### PR DESCRIPTION
**Pull request by kfarnung**
~~Needs to wait for https://github.com/Microsoft/ChakraCore/pull/5131 to land.~~
~~Needs to wait for https://github.com/Microsoft/ChakraCore/pull/5138 to land.~~

### CODE REVIEW

File: reference/JsGetPromiseResult.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Gets the result of a given promise object.
+
+### Syntax 
+```
+CHAKRA_API
+    JsGetPromiseResult(
+        _In_ JsValueRef *promise,
+        _Out_ JsPromiseState *statusCode);
```
Comment (MSLaguana) [inline reference/JsGetPromiseResult.md]: Looks like you've just copy/pasted from the other method; this should be a `JsValueRef` out parameter I believe.

**Comment by liminzhu** [comment]
Thanks for keeping the docs updated! 

Comments on documentation
1. Consider documenting behavior of `JsGetPromiseResult` when promise is pending (`JsErrorInvalidArgument` currently) and rejected (error object). At least the pending behavior isn't very obvious.

I hope it's not too late for comments on those APIs themselves
1. Most of our enums don't use _. Consider something like `JsPromiseStatePending`.
2. This is debatable - I'm not sure about `JsGetPromiseResult` returning `JsErrorInvalidArgument` from  pending promises. Intuitively, I only expect this if you throw in something other than promise. You can have a valid argument (a promise) but just catch it in a bad state. I'd suggest returning `JS_INVALID_REFERENCE` if you consider this a valid and successful operation or a new error code like `JsErrorPendingPromise` for a failed op.

**Comment by kfarnung** [comment]
@liminzhu I made the documentation changes to resolve your feedback, but it's pending another change to ChakraCore before this can land.

**Comment by kfarnung** [comment]
Opened https://github.com/Microsoft/ChakraCore/pull/5138

**Comment by kfarnung** [comment]
@liminzhu @MSLaguana Any other concerns with this documentation?

**Comment by liminzhu** [comment]
LGTM, thx!

**Review by liminzhu — APPROVED** [review]

**Review by MSLaguana — APPROVED** [review]

**Pull request merged**